core/esmf-aspect-model-java-generator/src/main/java/org/eclipse/esmf/aspectmodel/java/AspectModelJavaUtil.java

```diff
@@ -559,6 +560,13 @@ public static String genericClassSignature( final StructureElement element ) {
       return generics.isEmpty() ? "" : "<" + generics + ">";
    }
 
+   public static String generateClassName( final StructureElement element, final JavaCodeGenerationConfig config ) {
+      if ( ( !config.aspectPrefix().isBlank() || !config.aspectPostfix().isBlank() ) && element instanceof Aspect ) {
```

Comment:
I think this should work not only for Aspects, but also for other structure elements (i.e., Entities and AbstractEntities):
```suggestion
      if ( ( !config.aspectPrefix().isBlank() || !config.aspectPostfix().isBlank() ) && element.is( StructureElement.class ) ) {
```